contracts/DCAStrategies/DCAStrategies/DCAStrategiesPositionsHandler.sol

```diff
@@ -7,7 +7,45 @@ abstract contract DCAStrategiesPositionsHandler is IDCAStrategiesPositionsHandle
   // TODO: add function similar to this one https://github.com/Mean-Finance/dca-v2-core/blob/main/contracts/interfaces/IDCAHub.sol#L243
 
   /// @inheritdoc IDCAStrategiesPositionsHandler
-  function deposit(IDCAStrategies.DepositParams calldata parameters) external override returns (uint256) {}
+  function deposit(IDCAStrategies.DepositParams calldata parameters) external returns (uint256) {
+    IDCAStrategies.ShareOfToken[] memory _tokens = _getTokenShares(parameters.strategyId, parameters.version);
+    if (_tokens.length == 0) revert InvalidStrategy();
+
+    IERC20(parameters.from).transferFrom(msg.sender, address(this), parameters.amount);
```

Comment:
Let's use `safeTransferFrom` here, like [this](https://github.com/Mean-Finance/swappers/blob/main/solidity/contracts/SwapAdapter.sol#L31)